python/lsst/pipe/analysis/analysis.py

```diff
@@ -779,13 +787,21 @@ def plotAll(self, dataId, filenamer, log, enforcer=None, butler=None, camera=Non
                                stats=stats, hscRun=hscRun, matchRadius=matchRadius, zpLabel=zpLabel,
                                forcedStr=forcedStr)
 
-        self.plotSkyPosition(filenamer(dataId, description=self.shortName, style="sky-stars" + postFix),
-                             stats=stats, dataId=dataId, butler=butler, camera=camera, ccdList=ccdList,
-                             tractInfo=tractInfo, patchList=patchList, hscRun=hscRun, matchRadius=matchRadius,
-                             zpLabel=zpLabel, highlightList=highlightList, forcedStr=forcedStr,
-                             dataName="star")
+        if "all" in self.data:
+            self.plotSkyPosition(filenamer(dataId, description=self.shortName, style="sky-all" + postFix),
+                                 stats=stats, dataId=dataId, butler=butler, camera=camera, ccdList=ccdList,
+                                 tractInfo=tractInfo, patchList=patchList, hscRun=hscRun,
+                                 matchRadius=matchRadius, zpLabel=zpLabel, highlightList=highlightList,
+                                 forcedStr=forcedStr, dataName="all")
+
+        if "star" in self.data:
+            self.plotSkyPosition(filenamer(dataId, description=self.shortName, style="sky-stars" + postFix),
+                                 stats=stats, dataId=dataId, butler=butler, camera=camera, ccdList=ccdList,
+                                 tractInfo=tractInfo, patchList=patchList, hscRun=hscRun,
+                                 matchRadius=matchRadius, zpLabel=zpLabel, highlightList=highlightList,
+                                 forcedStr=forcedStr, dataName="star")
```

Comment:
Instead of duplicating plotSkyPosition it might be better to have style as a variable.

Reply:
I actually do want them to fire each time, but I simplified a bit by making a "kwargs" dict for the common arguments.